lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -293,11 +298,30 @@ sub log_entry {
     );
 }
 
-sub _stderr_is_tty{
+=head2 _process_data
+
+Process the data before printing out.
+
+Parameters: $self, $data
+Return: processed data
+
+=cut
+
+sub _process_data {
+    my ($self, $data) = @_;
+    $data = clone($data);
+    return $data if(numeric_level($data->{severity}) <= numeric_level('warn'));
+    # now severity > warn
+    return $data if $self->{log_level} >= numeric_level('debug');
```

Comment:
based on issue's info : 

> severity of logs > warning, will print stack information if asked log level is trace or debug

where did you check if the log level is `trace`?

Reply:
log level >= debug, it means log level is trace or debug:
```
8         trace
7         debug
6         info (inform)
5         notice
4         warning (warn)
3         error (err)
2         critical (crit, fatal)
1         alert
0         emergency

```